docs/reference/mapping/types/dense-vector.asciidoc

```diff
@@ -448,3 +448,72 @@ POST /my-bit-vectors/_search?filter_path=hits.hits
 }
 ----
 
+==== Updatable field type
+
+To better accommodate scaling and performance needs, updating the `type` setting in `index_options` is possible according to the following graph (jumps allowed):
+
+[source,txt]
+----
+flat --> int8_flat --> int4_flat --> hnsw --> int8_hnsw --> int4_hnsw
+----
+
+The following special cases apply:
+
+* updating `type` from `hnsw` to `int8_hnsw` is possible when:
+** the `m` setting is updated to the same or larger value
+** the `confidence_interval` setting is either unspecified in the original mapping or it doesn't get changed
+* updating `type` from `int8_hnsw` to `int8_hnsw` is possible when:
+** the `m` setting is updated to the same or larger value
+** the `confidence_interval` setting is either unspecified in the original mapping or it doesn't get updated
+* updating `type` from `int4_hnsw` to `int4_hnsw` is possible when:
+** the `m` setting is updated to the same or larger value
+** the `confidence_interval` setting is left untouched
```

Comment:
This reads weird. Maybe just indicate that the number of connections `m` must always increase, can never decrease and for scalar quantized formats the confidence interval must be consistent.

Reply:
thanks Ben, I like your suggestion.